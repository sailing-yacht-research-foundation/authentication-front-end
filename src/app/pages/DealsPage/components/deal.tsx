import React from 'react';
import { Row } from 'antd';
import styled from 'styled-components';
import DealLogoSvg from '../assets/dummy-logo.svg';
import { StyleConstants } from 'styles/StyleConstants';
import { Link } from 'app/components/Link';
import { media } from 'styles/media';

export const Deals = () => {
  return (
    <Row justify="center" align="top" style={{ minHeight: '100vh', marginTop: '100px', padding: '0 20px' }}>
      <DealWrapper>
        <DealInnerwrapper>
          <DealLogo src={DealLogoSvg} />
          <DealText>Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum</DealText>
          <Link to="/">Checkout this deal!</Link>
        </DealInnerwrapper>

        <DealInnerwrapper>
          <DealLogo src={DealLogoSvg} />
          <DealText>Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum</DealText>
          <Link to="/deals">Checkout this deal!</Link>
        </DealInnerwrapper>

        <DealInnerwrapper>
          <DealLogo src={DealLogoSvg} />
          <DealText>Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum</DealText>
          <Link to="/deals">Checkout this deal!</Link>
        </DealInnerwrapper>

        <DealInnerwrapper>
          <DealLogo src={DealLogoSvg} />
          <DealText>Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum</DealText>
          <Link to="/deals">Checkout this deal!</Link>
        </DealInnerwrapper>

        <DealInnerwrapper>
          <DealLogo src={DealLogoSvg} />
          <DealText>Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum</DealText>
          <Link to="/deals">Checkout this deal!</Link>
        </DealInnerwrapper>
      </DealWrapper>
    </Row>
  )
}

const DealWrapper = styled.div`
    display: flex;
    justify-content:center;
    flex-direction: column;

    ${media.medium`
      flex-direction: row;
    `}
`;

const DealInnerwrapper = styled.div`
  display: flex;
  flex-direction: column;
  max-width: 255px;
  align-items:center;
`;

const DealText = styled.p`
  font-family: ${StyleConstants.FONT_OPEN_SANS}
  text-align: center;
  font-size: 14px;
  font-weight: 400px;
  line-height: 28px;
  color: #737475;
  text-align: center;
`;

const DealLogo = styled.img`
    margin: 0 15px;
    height: 100px;
`;