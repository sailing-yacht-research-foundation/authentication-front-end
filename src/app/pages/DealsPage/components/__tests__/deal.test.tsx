import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { Deals } from '../deal';

const shallowRenderer = createRenderer();

describe('<Deals />', () => {
  it('should render the deals and matchs the snapshot', () => {
    shallowRenderer.render(<Deals />);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});