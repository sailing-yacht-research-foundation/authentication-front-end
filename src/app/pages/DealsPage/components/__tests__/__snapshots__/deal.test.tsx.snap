// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Deals /> should render the deals and matchs the snapshot 1`] = `
<Row
  align="top"
  justify="center"
  style={
    Object {
      "marginTop": "100px",
      "minHeight": "100vh",
      "padding": "0 20px",
    }
  }
>
  <styled.div>
    <styled.div>
      <styled.img
        src="dummy-logo.svg"
      />
      <styled.p>
        Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum
      </styled.p>
      <Link
        to="/"
      >
        Checkout this deal!
      </Link>
    </styled.div>
    <styled.div>
      <styled.img
        src="dummy-logo.svg"
      />
      <styled.p>
        Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum
      </styled.p>
      <Link
        to="/deals"
      >
        Checkout this deal!
      </Link>
    </styled.div>
    <styled.div>
      <styled.img
        src="dummy-logo.svg"
      />
      <styled.p>
        Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum
      </styled.p>
      <Link
        to="/deals"
      >
        Checkout this deal!
      </Link>
    </styled.div>
    <styled.div>
      <styled.img
        src="dummy-logo.svg"
      />
      <styled.p>
        Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum
      </styled.p>
      <Link
        to="/deals"
      >
        Checkout this deal!
      </Link>
    </styled.div>
    <styled.div>
      <styled.img
        src="dummy-logo.svg"
      />
      <styled.p>
        Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum
      </styled.p>
      <Link
        to="/deals"
      >
        Checkout this deal!
      </Link>
    </styled.div>
  </styled.div>
</Row>
`;
